plugins/experimental/log_requests/log_requests.c

```diff
@@ -0,0 +1,301 @@
+                                TS_HTTP_STATUS_PROXY_AUTHENTICATION_REQUIRED,
+                                TS_HTTP_STATUS_REQUEST_TIMEOUT,
+                                TS_HTTP_STATUS_CONFLICT,
+                                TS_HTTP_STATUS_GONE,
+                                TS_HTTP_STATUS_LENGTH_REQUIRED,
+                                TS_HTTP_STATUS_PRECONDITION_FAILED,
+                                TS_HTTP_STATUS_REQUEST_ENTITY_TOO_LARGE,
+                                TS_HTTP_STATUS_REQUEST_URI_TOO_LONG,
+                                TS_HTTP_STATUS_UNSUPPORTED_MEDIA_TYPE,
+                                TS_HTTP_STATUS_REQUESTED_RANGE_NOT_SATISFIABLE,
+                                TS_HTTP_STATUS_EXPECTATION_FAILED,
+                                TS_HTTP_STATUS_UNPROCESSABLE_ENTITY,
+                                TS_HTTP_STATUS_LOCKED,
+                                TS_HTTP_STATUS_FAILED_DEPENDENCY,
+                                TS_HTTP_STATUS_UPGRADE_REQUIRED,
+                                TS_HTTP_STATUS_PRECONDITION_REQUIRED,
+                                TS_HTTP_STATUS_TOO_MANY_REQUESTS,
+                                TS_HTTP_STATUS_REQUEST_HEADER_FIELDS_TOO_LARGE,
+                                TS_HTTP_STATUS_INTERNAL_SERVER_ERROR,
+                                TS_HTTP_STATUS_NOT_IMPLEMENTED,
+                                TS_HTTP_STATUS_BAD_GATEWAY,
+                                TS_HTTP_STATUS_SERVICE_UNAVAILABLE,
+                                TS_HTTP_STATUS_GATEWAY_TIMEOUT,
+                                TS_HTTP_STATUS_HTTPVER_NOT_SUPPORTED,
+                                TS_HTTP_STATUS_VARIANT_ALSO_NEGOTIATES,
+                                TS_HTTP_STATUS_INSUFFICIENT_STORAGE,
+                                TS_HTTP_STATUS_LOOP_DETECTED,
+                                TS_HTTP_STATUS_NOT_EXTENDED,
+                                TS_HTTP_STATUS_NETWORK_AUTHENTICATION_REQUIRED};
+
+static bool should_log(TSHttpTxn txnp);
+static void log_request_line(TSMBuffer bufp, TSMLoc loc, const char *output_header);
+static void log_response_status_line(TSMBuffer bufp, TSMLoc loc, const char *output_header);
+static void log_headers(TSMBuffer bufp, TSMLoc loc, const char *output_header);
+static void log_full_transaction(TSHttpTxn txnp);
+static int log_requests_plugin(TSCont contp ATS_UNUSED, TSEvent event, void *edata);
+
+static bool
+should_log(TSHttpTxn txnp)
+{
+  TSMBuffer txn_resp_bufp;
+  TSMLoc txn_resp_loc;
+  TSHttpStatus resp_status;
+
+  if (TSHttpTxnClientRespGet(txnp, &txn_resp_bufp, &txn_resp_loc) != TS_SUCCESS) {
+    TSError("[log_requests] Couldn't retrieve server response header.");
+    return false;
+  }
+
+  // if resp_status is blacklisted, don't log
+  resp_status = TSHttpHdrStatusGet(txn_resp_bufp, txn_resp_loc);
+  for (int i = 0; i < blacklist_size; ++i) {
+    if (resp_status == blacklist[i])
+      return false;
+  }
```

Comment:
Instead of the `for` loop this might be a bit cleaner. Although you should be using `std::vector` instead of raw int arrays.
```
if (std::any_of(blacklist, blacklist+blacklist_size, [=] (int code) { return code == resp_status; }))
  return false;
```